ravenframework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -88,10 +90,23 @@ def run(self, inputIn):
     """
     dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
     names = list(dataDict.keys())#
-    evaluation ={k: np.atleast_1d(val) for k, val in self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+
+    if 'time' in inputIn['Data'][0][-1].coords.keys():
+      PCMVersion = 'Snapshot'
+      print('Running Snapshot-PCM')
+      pivotParameter = self.pivotParameter
+      if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+        if 'dynamic' not in self.dynamicType: #self.model.dataType:
+          self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+      evaluation ={k: np.atleast_1d(val) for k, val in self._evaluate(PCMVersion, dataDict, **{'dataobjectNames': names}).items()}
+      evaluation[pivotParameter] = inputIn['Data'][0][-1]['time']
+    else:
+      PCMVersion = 'Static'
+      print('Running Static-PCM')
+      evaluation ={k: np.atleast_1d(val) for k, val in self._evaluate(PCMVersion, dataDict, **{'dataobjectNames': names}).items()}
```

Comment:
I think you can use self.dynamic variable to indicate whether it is snapshot or static, and I would suggest you do not modify the function API for _evaluate method, since you can use self.dynamic in the _evaluate method to indicate the pcm version

Reply:
The API function for _evaluate method has been simplified. As snapshot PCM needs pivotParameter, an IF statement to obtain the values for pivotParameter is kept.